atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolTest.java

```diff
@@ -687,4 +767,30 @@ private void setupHostsWithInconsistentTopology(CassandraServer... cassandraServ
     private void refreshPool() {
         deterministicExecutor.tick(POOL_REFRESH_INTERVAL_SECONDS, TimeUnit.SECONDS);
     }
+
+    private void assertEmptyConnectionPoolExceptionThrown(
+            ThrowingCallable throwingCallable,
+            Set<CassandraServer> serversToAdd,
+            Set<CassandraServer> previousCassandraServers) {
+        assertThatLoggableExceptionThrownBy(throwingCallable)
+                .isInstanceOf(SafeIllegalStateException.class)
+                .hasMessageContaining("No servers were successfully added to the pool.")
+                .hasExactlyArgs(
+                        SafeArg.of("serversToAdd", CassandraLogHelper.collectionOfHosts(serversToAdd)),
+                        SafeArg.of(
+                                "previousCassandraServers",
+                                CassandraLogHelper.collectionOfHosts(previousCassandraServers)));
+        assertEmptyPoolMetricIsEqualTo(1);
+    }
+
+    private void assertEmptyPoolMetricIsEqualTo(int expected) {
+        assertThat(taggedMetricRegistry
+                        .counter(MetricName.builder()
+                                .safeName(MetricRegistry.name(
```

Comment:
this is yuck, but the other option is re-writing a bit of this class and exposing `CassandraClientPoolMetrics`. Happy to do this, but we had not done this explicitly previously (see the pooling metrics -- it uses the tagged registry), so biased towards that.

Reply:
makes sense!